src/Resources/contao/classes/Backend.php

```diff
@@ -413,8 +415,25 @@ protected function getBackendModule($module)
 		// Custom action (if key is not defined in config.php the default action will be called)
 		elseif (\Input::get('key') && isset($arrModule[\Input::get('key')]))
 		{
-			$objCallback = new $arrModule[\Input::get('key')][0]();
-			$this->Template->main .= $objCallback->{$arrModule[\Input::get('key')][1]}($dc);
+			$container  = static::getContainer();
+			$className  = $arrModule[\Input::get('key')][0];
+			$methodName = $arrModule[\Input::get('key')][1];
+
+			// Use the controller service
+			if (!class_exists($className) && $container->has($className)) {
+				$request = $container->get('request_stack')->getCurrentRequest();
+				$request->attributes->set('id', $dc->id);
+				$request->attributes->set('table', $dc->table);
+
+				/** @var Response $response */
+				$response = $container->get($className)->$methodName($request);
+
+				$this->Template->main .= $response->getContent();
+			} else {
```

Comment:
I'd add a `// BC` comment here and maybe even a `trigger_error()` for deprecated usage as there's now a proper way of implemnting this :)